funnel/models/event.py

```diff
@@ -52,6 +53,25 @@ class Event(BaseMixin, db.Model):
     ticket_types = db.relationship("TicketType", secondary=event_ticket_type, lazy='dynamic')
     participants = association_proxy('attendees', 'participant')
 
+    @classmethod
+    def get_or_create(cls, name, space):
+        event = cls.query.filter_by(name=name, proposal_space=space).first()
+        if not event:
+            event = cls(name=name, proposal_space=space)
+            db.session.add(event)
+            db.session.commit()
+        return event
+
+    @classmethod
+    def sync_from_list(cls, event_list, space):
+        for event_dict in event_list:
+            event = cls.get_or_create(event_dict.get('name').decode('utf-8'), space)
```

Comment:
Don't call `.decode('utf-8')` here. Clean up data before it is passed to the model.